grails-app/services/org/pih/warehouse/inventory/RequisitionStatusTransitionEventService.groovy

```diff
@@ -18,8 +19,9 @@ class RequisitionStatusTransitionEventService implements ApplicationListener<Req
     NotificationService notificationService
 
     void onApplicationEvent(RequisitionStatusTransitionEvent event) {
-        if (event.requisition.shouldSendApprovalNotification()) {
-            notificationService.publishRequisitionStatusTransitionNotifications(event.requisition)
+        Requisition requisition = Requisition.get(event.requisition.id)
```

Comment:
My conscience wouldn't allow me to leave this without a comment even though we agreed on a similar solution somewhere, so:
1. If the problem with the LazyInitializationException happens here, it should be enough to add `@Transactional` annotation for this service
2. If it happens in the view (which I assume happens), I'd prefer to create a data service method that would eagerly fetch the events using `@Join` annotation instead of fetching the requisition twice.

Reply:
yes, I agree a comment should be added.
By the way this is grails 1 so `@Join` services are probably not available to us here

Reply:
@drodzewicz 

> yes, I agree a comment should be added. 
I didn't mean to add a comment in the code, I just meant that my conscience wouldn't allow me to leave this **PR** without the comment :laughing: 

> By the way this is grails 1 so @Join services are probably not available to us here 
Oh, is it?..... then I apologize, but I'd consider adding a TODO to change it to the data service method I proposed in Grails 3, if everyone agrees on my preposition

Reply:
I think we should add the TODO for Grails 3

Reply:
I think we should always pass the ID rather than the object to these events (see RefreshProductAvailabilityEvent) because there's no guarantee the event handler will be executed synchronously. We should treat them similar to Quartz jobs.